src/exports.c

```diff
@@ -123,6 +207,78 @@ static void _obj_free(void **obj)
     *obj = NULL;
 }
 
+static bool
+_action_store(UNUSED bool filter_ret, UNUSED json_object *found,
+    UNUSED Needles current)
+{
+    return true;
+}
+
+static bool
+_action_store_true(bool filter_ret, UNUSED json_object *found,
+    UNUSED Needles current)
+{
+    if(filter_ret)
+        return true;
+    return false;
+}
+
+static bool
+_action_discard_false(bool filter_ret, UNUSED json_object *found,
+    UNUSED Needles current)
+{
+    if(filter_ret)
+        return true;
+    return false;
+}
+
+static bool
+_action_discard_true(bool filter_ret, UNUSED json_object *found,
+    UNUSED Needles current)
+{
+    if(filter_ret)
+        return false;
+    return true;
+}
+
+static bool
+_filter_match(int jpointer, json_object *found,
+    UNUSED Needles current)
+{
+    if(jpointer != 0) // no data to match against
+        return false;
+    if(strcmp(json_object_get_string(found), current->filter_data) == 0)
```

Comment:
This can happen if "found" is of json_type_null (e.g. NULL). I've checked against that case. We now should at least get a reference to an empty string.